File: NeuralNet/Regularization/L2Regularization.cs
namespace NeuralNet {
	public sealed class L2Regularization : Regularization {
		public L2Regularization(float regularizationFactor) : base(regularizationFactor) {}

		public override float GetDerivative(float value) {
			return RegularizationFactor*value;
		}
	}
}

File: NeuralNetNative/L2Regularization.cpp
#define NEURALNETNATIVEAPI
#include "L2Regularization.h"

namespace NeuralNetNative {
	L2Regularization::L2Regularization(float regularizationFactor) : Regularization(regularizationFactor) {
	}

	float L2Regularization::GetDerivative(float value) {
		return RegularizationFactor*value;
	}
}